server/src/main/java/io/crate/types/TypeSignature.java

```diff
@@ -195,7 +121,7 @@ public String getBaseTypeName() {
                     var valTypeSignature = parameters.get(i + 1);
                     if (valTypeSignature instanceof ParameterTypeSignature) {
                         var innerTypeName = ((ParameterTypeSignature) valTypeSignature).parameterName();
-                        builder.setInnerType(innerTypeName, valTypeSignature.createType());
+                        builder.setInnerType(unEscapeDataTypeValues(innerTypeName), valTypeSignature.createType());
```

Comment:
What do you think of having a `unescapedParameterName()` property on the type signature and change `parameterName()` to `escapedParameterName()`